### PR TITLE
Added get/setElementAngularVelocity

### PR DESCRIPTION
**Pull request by lex128**
Patch for https://bugs.mtasa.com/view.php?id=9374

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 1]
```diff
@@ -638,6 +643,62 @@ int CLuaElementDefs::OOP_GetElementVelocity ( lua_State* luaVM )
...
+{
+    // Verify the argument
+    CClientEntity* pEntity = NULL;
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadUserData ( pEntity );
+
+    if ( !argStream.HasErrors () )
+    {
+        // Grab the element, verify it
+        if ( pEntity )
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp]: "verifications" are done by `CScriptArgReader` now, so you don't need to manually do LogBadPointer.

File: Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 2]
```diff
@@ -2062,6 +2123,38 @@ int CLuaElementDefs::SetElementVelocity ( lua_State* luaVM )
...
+    CVector vecTurnVelocity;
+
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadUserData ( pEntity );
+    argStream.ReadVector3D ( vecTurnVelocity );
+
+    // Verify the arguments
+    if ( !argStream.HasErrors () )
+    {
+        if ( pEntity )
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp]: This needs changing too (see [first comment](https://github.com/multitheftauto/mtasa-blue/pull/73/files#r78756626))

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 3]
```diff
@@ -1427,6 +1447,46 @@ bool CStaticFunctionDefinitions::SetElementVelocity ( CElement* pElement, const
...
+    assert ( pElement );
+    RUN_CHILDREN( SetElementTurnVelocity ( *iter, vecTurnVelocity ) )
+
+    int iType = pElement->GetType ();
+    switch ( iType )
+    {
+        case CElement::PED:
+        case CElement::PLAYER:
+        {
+            // TODO
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1415]: Client side only?

**Comment by qaisjp** [comment]
I haven't tested this yet, but it looks good so far.

You may also want to add a commit to deprecate setVehicleTurnVelocity and rewrite it to setElementTurnVelocity. Refer to https://github.com/multitheftauto/mtasa-blue/commit/6b3aee3f1b1a904bd754cd9486fd4ac16c0d51dd for help on this (there are separate lists for client/server).

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp (hunk 3 above)
Comment (lex128) [reply Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1415]: While setting turn speed only on the server side for peds, without saving/getting

Comment (qaisjp) [reply Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1415]: Sorry, please can you rephrase?

Comment (lex128) [reply Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1415]: Because now the server side synchronizes only move speed of peds, without turn speed. And not synchronized objects at all. I think it can be done later. Sorry for my bad english 😊

**Comment by eAi** [comment]
Wouldn't get/setElementAngularVelocity be a better name?

**Review by qaisjp — DISMISSED** [review]
Please rename to get/setElementAngularVelocity

**Comment by Necktrox** [comment]
Applying the angular velocity on peds cause weird side-effects, which might not be desirable: https://www.youtube.com/watch?v=_eiS_UA_DQU

Applying the angular velocity on objects is weird, because they don't work unless the object has any kind of positional velocity on them (a value of 0.0001 would be enough) and the angular velocity decreases after time, but positional velocity doesn't.

I need feedback on this.

**Comment by lex128** [comment]
@Necktrox with high velocity values usually something cause weird effects, mostly with upside peds.

As we known the objects behaves as frozen when their linear velocity is zero. So do you suggest to apply a bit linear velocity when applying some angular velocity for the defrost effect?

**Comment by Necktrox** [comment]
I personally would rather prefer a patch to make it possible to apply angular velocity without any kind of positional velocity and applying linear velocity (if it's zero) beforehand is a dirty hack.

**Comment by lex128** [comment]
@qaisjp renamed

File: Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 4]
```diff
@@ -109,7 +111,6 @@ void CLuaElementDefs::AddClass(lua_State* luaVM)
     lua_classfunction(luaVM, "removeData", "removeElementData");
 
     lua_classfunction(luaVM, "setParent", "setElementParent");
-    lua_classfunction(luaVM, "setVelocity", "setElementVelocity");
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:114]: 👍 nice catch with the erroneous extra `setVelocity` function

**Review by qaisjp — DISMISSED** [review]
Code looks good to me. I'm happy for this to be *squash* merged once it has been tested.
